sunpy/visualization/animator/wcs.py

```diff
@@ -0,0 +1,241 @@
+            self.plot_dimensionality = 2
+
+        if self.plot_dimensionality == 1:
+            try:
+                from astropy.visualization.wcsaxes.frame import RectangularFrame1D
+            except ImportError as e:
+                raise ImportError("Astropy 4.0 must be installed to do line plotting with WCSAxes.") from e
+
+        self.naxis = data.ndim
+        self.num_sliders = self.naxis - self.plot_dimensionality
+        self.slices_wcsaxes = list(slices)
+        self.wcs = wcs
+        self.coord_params = coord_params
+        self.ylim = ylim
+        self.ylabel = ylabel
+
+        super().__init__(data, image_axes=image_axes, axis_ranges=None, **kwargs)
+
+    def _sanitize_axis_ranges(self, *args):
+        """
+        This overrides the behaviour of ArrayAnimator to generate axis_ranges
+        based on the WCS.
+        """
+
+        def partial_pixel_to_world(pixel_dimension, pixel_coord):
+            """
+            Return the world coordinate along one axis, if it is only
+            correlated to that axis.
+            """
+            wcs_dimension = self.wcs.pixel_n_dim - pixel_dimension - 1
+            corr = self.wcs.axis_correlation_matrix[:, wcs_dimension]
+
+            # If more than one world axis is linked to this dimension we can't
+            # display the world coordinate because we have no way of picking,
+            # so we just display pixel index.
+            if len(np.nonzero(corr)[0]) != 1:
+                return pixel_coord
+
+            # We know that the coordinate we care about is independent of the
+            # other axes, so we can set the pixel coordinates to 0.
+            coords = [0] * self.wcs.pixel_n_dim
+            coords[wcs_dimension] = pixel_coord
+            wc = self.wcs.pixel_to_world_values(*coords)[wcs_dimension]
+            return u.Quantity(wc, unit=self.wcs.world_axis_units[wcs_dimension])
+
+
+        axis_ranges = [None] * self.wcs.pixel_n_dim
+        for i in self.slider_axes:
+            axis_ranges[i] = partial(partial_pixel_to_world, i)
+
+        return axis_ranges, None
+
+
+    def _apply_coord_params(self):
+        if self.coord_params is None:
+            return
+
+        for coord_name in self.coord_params:
+            coord = self.axes.coords[coord_name]
+            params = self.coord_params[coord_name]
+
+            format_unit = params.get("format_unit", None)
+            if format_unit:
+                coord.set_format_unit(format_unit)
+
+            major_formatter = params.get("major_formatter", None)
+            if major_formatter:
+                coord.set_major_formatter(major_formatter)
+
+            axislabel = params.get("axislabel", None)
+            if axislabel:
+                coord.set_axislabel(axislabel)
+
+            grid = params.get("grid", None)
+            if grid is not None:
+                if not isinstance(grid, dict):
+                    grid = {}
+                coord.grid(**grid)
```

Comment:
So the changes/actions of ```coord``` doesn't have to be done to the version attached to ```self```?

Reply:
No, coord is attached to `self.axes` and coord is just the element in the dict, so this will happily work by reference.

Reply:
So ```coord``` is a shallow copy of something attached to ```self.axes``` not a new deepcopy?

Reply:
it's not even a copy really, it's just a variable name pointing at the object in the dict.